bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/WebDataDownloadUtil.java

```diff
@@ -0,0 +1,148 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.net;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Some utility functions related to WEB data download.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+public class WebDataDownloadUtil {
+
+    private WebDataDownloadUtil() {
+    }
+
+    /**
+     * Download the image data from an URL.
+     *
+     * If content type is not found in the headers, the data is scanned to determine the content type.
+     *
+     * @param url the URL of the image to be downloaded
+     * @return a RawType object containing the image, null if the content type could not be found or the content type is
+     *         not an image
+     */
+    public static RawType downloadImage(URL url) {
+        return WebDataDownloadUtil.downloadImage(url, true, -1);
+    }
+
+    /**
+     * Download the image data from an URL.
+     *
+     * @param url the URL of the image to be downloaded
+     * @param scanTypeInContent true to allow the scan of data to determine the content type if not found in the headers
+     * @param maxContentLength the maximum data size in bytes to trigger the download; any negative value to ignore the
+     *            data size
+     * @return a RawType object containing the image, null if the content type could not be found or the content type is
+     *         not an image or the data size is too big
+     */
+    public static RawType downloadImage(URL url, boolean scanTypeInContent, long maxContentLength) {
+        return WebDataDownloadUtil.downloadData(url, "image/.*", scanTypeInContent, maxContentLength);
+    }
+
+    /**
+     * Download the data from an URL.
+     *
+     * @param url the URL of the data to be downloaded
+     * @param contentTypeRegex the REGEX the content type must match; null to ignore the content type
+     * @param scanTypeInContent true to allow the scan of data to determine the content type if not found in the headers
+     * @param maxContentLength the maximum data size in bytes to trigger the download; any negative value to ignore the
+     *            data size
+     * @return a RawType object containing the downloaded data, null if the content type does not match the expected
+     *         type or the data size is too big
+     */
+    public static RawType downloadData(URL url, String contentTypeRegex, boolean scanTypeInContent,
+            long maxContentLength) {
+
+        String contentType = RawType.DEFAULT_MIME_TYPE;
+        if (contentTypeRegex != null) {
+            contentType = WebDataDownloadUtil.getContentType(url, scanTypeInContent, maxContentLength);
```

Comment:
Yes, that's true. It is necessary when you scan the content to determine the content type. But when this scan is not required, I agree that we should avoid opening the connection twice. That means having only one function.